lms/djangoapps/courseware/courses.py

```diff
@@ -370,7 +370,10 @@ def get_course_info_section(request, user, course, section_key):
     html = ''
     if info_module is not None:
         try:
-            html = info_module.render(STUDENT_VIEW).content.strip()
+            if user.is_authenticated():
+                html = info_module.render(STUDENT_VIEW).content.strip()
+            else:
+                html = info_module.render(PUBLIC_VIEW).content.strip()
```

Comment:
Is it okay to do this without checking the waffle flag and course settings?

Reply:
I found that this method is only called by the line:
https://github.com/edx/edx-platform/blob/master/openedx/features/course_experience/views/course_home.py#L104

The waffle flag and course settings are checked before calling the above function.